README.md
## Getting daily currency, This automated script is using [jq](https://stedolan.github.io/jq/)\n## Last Update:  2020-02-07 à 08:10:40\n # Monaie de base : 1 "EUR" 💶 \n # Equivaut en 🇹🇳 à :  3.105122 \n # Equivaut en 💵 à : 1.097701\n # Equivaut en 🐱‍💻 à :  0.000112


CHANGELOG.md

## [2020-01-30]

## [2020-01-29]

## [2020-01-24]

## [2020-01-23]

## [2020-01-18]

## [2020-01-14]

## [2020-01-13]

## [2020-01-12]

## [2020-01-11]

## [2020-01-09]

## [2020-01-08]

## [2020-01-02]

## [2019-12-29]

## [2019-12-28]

## [2019-12-27]

## [2019-12-26]

## [2019-12-25]

## [2019-12-24]

## [2019-12-23]

## [2019-12-22]

## [2019-12-21]

## [2019-12-19]

## [2019-12-18]

## [2019-12-17]

## [2019-12-16]

## [2019-12-15]

## [2019-12-14]

## [2019-12-13]

## [2019-12-12]

## [2019-12-11]

## [2019-12-10]

## [2019-12-09]

## [2019-12-08]

## [2019-12-07]

## [2019-12-06]

## [2019-12-05]

## [2019-12-04]

## [2019-12-03]

## [2019-12-02]

## [2019-12-01]

## [2019-11-30]

## [2019-11-29]

## [2019-11-28]

## [2019-11-27]

## [2019-11-26]

## [2019-11-25]

## [2019-11-24]

## [2019-11-14]

## [2019-11-03]

## [2019-11-02]

## [2019-11-01]

## [2019-10-31]

## [2019-10-30]

## [2019-10-29]

## [2019-10-28]

## [2019-10-27]

## [2019-10-26]

## [2019-10-25]

## [2019-10-24]

## [2019-10-23]

## [2019-10-22]

## [2019-10-21]

## [2019-10-20]

## [2019-10-19]

## [2019-10-18]

## [2019-10-17]

## [2019-10-16]

## [2019-10-15]

## [2019-10-14]

## [2019-10-13]

## [2019-10-12]

## [2019-10-11]

## [2019-10-10]

## [2019-10-09]

## [2019-10-08]

## [2019-10-07]

## [2019-10-06]

## [2019-10-05]

## [2019-10-04]

## [2019-10-03]

## [2019-10-02]

## [2019-10-01]

## [2019-09-30]

## [2019-09-29]

## [2019-09-28]

## [2019-09-27]

## [2019-09-26]

## [2019-09-25]

## [2019-09-22]

## [2019-09-21]

## [2019-09-20]

## [2019-09-19]

## [2019-09-18]

## [2019-09-17]

## [2019-09-16]

## [2019-09-15]

## [2019-09-14]

## [2019-09-13]


test.sh
#!/bin/sh

FORMAT=" %s - $COMMITREF "
COMMITREF="[%h](../../commit/%h)"
#echo "# CHANGELOG" >> $CHANGELOG_FILE
NEXT=$(date +%F)
echo "CHANGELOG"
echo "$DATE"
echo ----------------------
git log --no-merges --format="%cd" --date=short | sort -u -r | while read DATE ; do
    echo
    #echo "<h3> :shipit: [$DATE] :shipit: </h3>"
    echo "## [$DATE]"
git log  --format="$FORMAT <br>" --since="$DATE 00:00:00:" --until="$DATE 24:00:00"  | sort -k2 -n  | less  | awk -F"\t" '/[NEW|IMPROVE|FIX|RELEASE|DOC|TEST]/ ' |  awk  '
prev!=$1  { prev=$1 ; printf "%s>%s <br>", NR==$1? ":" : ORS, $1 $2 , ORS }
prev==$1 { for (i=3;i<=NF; i++){
   printf "%s%s", $i, i==NF ? ORS : OFS
   
   } 
   
   }
'  | awk -
NEXT=$DATE 
done > CHANGELOG.md



auto.sh
#!/bin/sh


. /home/taieb/Bureau/Bash/autobotnews/secret
changeEmojiDependingtohour(){
    a=5
    b=10
    c=15
    d=23
    messageEmoji=""
    if [ $(date +%H) -le "$a" ]
    then
       echo ':package: NEW: '
         # echo $messageEmoji
    elif [ $(date +%H) -le "$b" ]
    then
        echo ':bug: FIX: '
    elif [ $(date +%H) -le "$c" ]
    then
     echo ':rocket: RELEASE: '

    elif [ $(date +%H) -le "$d" ]
    then echo ':white_check_mark: TEST: '
    fi
    }



getInfo(){
    
    result= curl -s  "http://data.fixer.io/api/latest?access_key=$APIKEY&format=1/" | jq '.timestamp,.base,.rates.TND,.rates.USD,.rates.BTC'
    echo $result
    
}
# getTime(){
#     now=$(date '+%F à %H:%M:%S')
#     echo $now
# }
extract(){
    now=$(date '+%F à %H:%M:%S')
    
    timeupdate=$(head -n 1 currency.json)
    base=$(sed '2q;d' /home/taieb/Bureau/Bash/autobotnews/currency.json)
    tnd=$(sed '3q;d' /home/taieb/Bureau/Bash/autobotnews/currency.json)
    usd=$(sed '4q;d' /home/taieb/Bureau/Bash/autobotnews/currency.json)
    btc=$(sed '5q;d' /home/taieb/Bureau/Bash/autobotnews/currency.json)
    # echo $base
    
 echo  "## Getting daily currency, This automated script is using [jq](https://stedolan.github.io/jq/)\n""## Last Update: " $now"\n ""# Monaie de base : 1" $base 💶 "\n" "# Equivaut en 🇹🇳 à : " $tnd "\n" "# Equivaut en 💵 à :" $usd"\n" "# Equivaut en 🐱‍💻 à : " $btc > /home/taieb/Bureau/Bash/autobotnews/README.md    
}

push(){
    
    cd "/home/taieb/Bureau/Bash/autobotnews/"
    git add .
    git commit -m   " $(changeEmojiDependingtohour) Currency updated @ $now"
    git push
}


changeEmojiDependingtohour
getInfo > /home/taieb/Bureau/Bash/autobotnews/currency.json
# getTime > /home/taieb/Bureau/Bash/autobotnews/time.txt
extract
push


changlog.sh
#!/bin/bash
# Generates changelog day by day


RED="\033[1;31m"
GREEN="\033[0;32m"
YELLOW="\033[1;33m"
BLUE="\033[1;34m"
PURPLE="\033[1;35m"
CYAN="\033[1;36m"
WHITE="\033[1;37m"
RESET="\033[0m"
STARTING_DATE="2019-12-16 00:00:00"
getVersion(){
        BASE_STRING=`cat VERSION`
    BASE_LIST=(`echo $BASE_STRING | tr '.' ' '`)
    V_MAJOR=${BASE_LIST[0]}
    V_MINOR=${BASE_LIST[1]}
    V_PATCH=${BASE_LIST[2]}
    echo -e "${NOTICE_FLAG} Current version: ${WHITE}$BASE_STRING"
    echo -e "${NOTICE_FLAG} Latest commit hash: ${WHITE}$LATEST_HASH"
    V_MINOR=$((V_MINOR + 1))
    V_PATCH=0
    SUGGESTED_VERSION="$V_MAJOR.$V_MINOR.$V_PATCH"
    echo -ne "${QUESTION_FLAG} ${CYAN}Enter a version number [${WHITE}$SUGGESTED_VERSION${CYAN}]: "
    read INPUT_STRING
    if [ "$INPUT_STRING" = "" ]; then
        INPUT_STRING=$SUGGESTED_VERSION
    fi
    echo -e "${NOTICE_FLAG} Will set new version to be ${WHITE}$INPUT_STRING"
    echo $INPUT_STRING > VERSION
}

#cat temp.md | awk '$2!=prev{print (prev=$2 ":") }  {print $s}'   > test.md
CHANGELOG_FILE=CHANGELOG.md

deploy(){
COMMITREF="[%h](../../commit/%h)"
FORMAT=" %s - $COMMITREF "

echo "# CHANGELOG" >> $CHANGELOG_FILE
NEXT=$(date +%F)
echo "CHANGELOG"
echo "$DATE"
echo ----------------------
git log --no-merges --format="%cd" --date=short --since="2019-12-29 00:00:00"  --until="$DATE 24:00:00"  | sort -u -r | while read DATE ; do
    echo
    #echo "<h3> :shipit: [$DATE] :shipit: </h3>"
    echo "## [$DATE]"
       # GIT_PAGER=cat git log --no-merges --format=">$FORMAT <br>" --after="$DATE 00:00:00" --until="$DATE 24:00:00"  |   sort -g |  sed -n -e '/📦/{p;n;}' -e '/🐛/{p;n;}' -e '/🚀/{p;n;}' -e '/📦/{p;n;}' -e '/✅/{p;n;}' 
    GIT_PAGER=cat git log  --no-merges  --format="$FORMAT <br>" --since="2019-12-29 00:00:00"  --until="$DATE 24:00:00" |   sort -g |  sed  '
s/🐛FIX:/:bug: FIX: /g ; 
s/👌IMPROVE:/:ok_hand: IMPROVE: /g;
s/:book:IMPROVE:Improve/:book: IMPROVE: /g;
s/📦NEW/:package: NEW:/g;
s/🚀RELEASE:/:rocket: RELEASE: /g;
s/✅TEST/:white_check_mark: TEST: /g;
s/📖DOC:/:book: DOC: /g' |   sort -k2 -n  | less  | awk -F"\t" '/[NEW|IMPROVE|FIX|RELEASE|DOC|TEST]/ ' |  awk  '
prev!=$1  { prev=$1 ; printf "%s>%s <br>", NR==$1? ":" : ORS, $1 $2 , ORS }
prev==$1 { for (i=3;i<=NF; i++){
   printf "%s%s", $i, i==NF ? ORS : OFS
   
   } 
   
   }
'  

#    NEXT=$DATE 

done >  CHANGELOG.md 

#cat diff temp.md CHANGELOG.md

}


if [ "$CHANGELOG_FILE"  != "" ]; then
  rm -rf $CHANGELOG_FILE
  touch $CHANGELOG_FILE
fi
: '
echo "### Select the Operation you want to perform ###"
echo "  1)Versioning & Deploy "
echo "  2)Deploy"

read n
case $n in
  1) getVersion; deploy;;
  2) deploy;;
esac
'

#getVersion
deploy